openhexa/sdk/workspaces/current_workspace.py

```diff
@@ -203,45 +229,45 @@ def get_connection(
                 raise ValueError(f"Connection {identifier} does not exist.")
 
             for d in data["fields"]:
-                fields[d.get("code")] = d.get("value")
+                connection_fields[d.get("code")] = d.get("value")
 
-            connection_type = data["type"]
+            connection_type = data["type"].upper()
         else:
             try:
                 env_variable_prefix = stringcase.constcase(identifier.lower())
-                for key, val in os.environ.items():
-                    if key.startswith(f"{env_variable_prefix}_"):
-                        field_name = key[len(f"{env_variable_prefix}_") :].lower()
-                        fields[field_name] = val
-
-                connection_type = os.environ[f"{env_variable_prefix}"]
+                connection_type = os.environ[f"{env_variable_prefix}"].upper()
+                connection_fields = self._get_local_connection_fields(env_variable_prefix)
             except KeyError:
                 raise ValueError
 
         if not connection_type:
             raise ValueError(f"Connection {identifier} does not exist.")
 
-        connection_type = connection_type.upper()
-        if connection_type in ConnectionClasses.keys():
-            if connection_type == "S3":
-                secret_access_key = fields.pop("access_key_secret")
-                return S3Connection(secret_access_key=secret_access_key, **fields)
-
-            if connection_type == "POSTGRESQL":
-                db_name = fields.pop("db_name")
-                port = int(fields.pop("port"))
-                return PostgreSQLConnection(database_name=db_name, port=port, **fields)
-
-            if connection_type == "CUSTOM":
-                dataclass = make_dataclass(
-                    stringcase.pascalcase(identifier),
-                    fields.keys(),
-                    bases=(CustomConnection,),
-                    repr=False,
-                )
-                return dataclass(**fields)
-
-            return ConnectionClasses[connection_type](**fields)
+        # In connected mode (API call) the secret_access_key field and db_name name are
+        # different from the offline ones
+        if connection_type == "S3":
+            secret_access_key = (
+                connection_fields.pop("access_key_secret")
+                if self._connected
+                else connection_fields.pop("secret_access_key")
+            )
+            return S3Connection(secret_access_key=secret_access_key, **connection_fields)
+
+        if connection_type == "POSTGRESQL":
+            db_name = connection_fields.pop("db_name") if self._connected else connection_fields.pop("database_name")
```

Comment:
If you build the `connection_fields` yourself, why do you not put the same key instead of adding this kind of condition ?